src/components/login/login.tsx

```diff
@@ -68,8 +78,15 @@ const Login: React.FC<LoginProps> = (props: LoginProps) => {
                   : ""
               }
               input={username}
-              onBlur={() => setFormDirty(true)}
-              onChange={setUsername}
+              onChange={(username: string) => {
+                setUsername(username);
+                setFormError("");
+              }}
+              onBlur={() => {
+                if (password) {
+                  setFormDirty(true);
+                }
```

Comment:
only show "enter username" error if the user has entered a password and left username blank, otherwise its too early to show an error.